rfc/20250905-module-package-metadata.md

```diff
@@ -0,0 +1,53 @@
+# Module Package Metadata 
+
+This RFC introduces the concept of Module Package Metadata to allow module authors to clearly define information about the modules defined within the package.
+
+This is primarilly motivated by the [Module Installation Specifications RFC](../rfc/20250905-module-installation-specifications.md) and it's primary issue https://github.com/opentofu/opentofu/issues/1086 
+
+## Proposed Solution
+
+To define the initial format for a module-package.meta.hcl file in the root of installed module packages. This would be useful to specify that a module is designed for use in OpenTofu and potentially opt-into new functionality.
```

Comment:
Does this really make sense at the root of an entire module package? I have concerns that for people with large module packages that contain a ton of subdirectories, if you were to introduce this concept some people would want a single metadata file for the entire module package and others would want a separate metadata file per subdirectory. If this were to be implemented, I would want both of those dynamics to be supported but that then raises a lot of complexity. Do you inherit and merge the parent metadata file? Do you just grab the first one in the filesystem hierarchy?